### DISCUSSION TITLE
Development through FDS

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#13944] [Q&A]
Dear team,
After I read the [FDS Road Map](https://github.com/firemodels/fds/wiki/FDS-Road-Map) agian. I decide to seek for help. It's still me, planning to simulate the process of liquid sodium combustion using FDS. I am a postgraduate student currently working on a simulation of liquid sodium combustion. In a nuclear reactor system, liquid sodium usually operates at more than 500°C. In the event of an accident, such as a rupture of a pipeline that causes sodium to spew out and come into contact with air, a sodium fire can occur. It is categorized into spray sodium fire and pool sodium fire. Simply say, a spray sodium fire is a small amount of spray, most of which burns in the form of liquid droplets, causing the ambient temperature and pressure to rise in a short period of time. Pool sodium fire that is mostly liquid sodium flow to the ground, in the form of liquid pools of evaporation combustion, longer duration. Aerosols such as sodium oxides are released during all of them.
I would like to use FDS and synthesize the existing functions of FDS to add liquid sodium oxidation model, combustion model, aerosol deposition model, size distribution model, etc. to it, so as to simulate the process in line with the liquid sodium leakage in a nuclear reactor.
Do you have any good suggestions?
Really hope to recieve your kind reply.

LI

**Comment by drjfloyd** [comment]
In FDS you can specify a solid phase reaction that consumes oxygen from the gas, you can define a reaciton that burns sodium vapor, both the solid phase reaction and the gas phase reaction can produe an aerosol, there is an aerosol deposiiton model, and aerosols can have a size distribution. You should read the Technical Reference Guide and then study the sections of the User's Guide and Verificiation Guide for the phenomena of interest to you. Including reviewing the various example files. Then I suggest you start with very simple test cases as you develop your inputs before attempting to model the full scale problem.

**Reply by lihy213** [reply]
Yes, I have successfully simulated liquid sodium combustion before by using a liquid pyrolysis model in the form of input cards. However, by setting up an arithmetic example similar to the experiment given the parameters of the input card, e.g., heat of combustion, sodium physical properties, etc., the simulation results were quite different from the experiment.

**Reply by drjfloyd** [reply]
Not matching an experiment can be any number of reasons: grid resolution, error in inputs describing the experiment, some key detail of the experiment missing from the experimental report, uncertainty in experimental measurements, uncertainty in model predictions, etc.

First make sure you have tested your inputs with simple test cases and that you are satisfied that in simple cases where boundary conditions are clearly specified that you are getting the results you expect. If you have done that, next is to carefully check the inputs for your full scale case. It is all too easy to transpose digits, mistype a digit, etc. in an input. If you are sure all your inputs are correct, then you need to make sure you have suffiicient outputs that you can determine why you are not getting the results you expect. This is what it means to be doing research.  If you reach a point where you think you have found an error in what FDS is doing or have something you cannot explain then provide us with an input and a clear description of the specific problem.

**Reply by lihy213** [reply]
I will double check my case and ask for your advice thereafter.
